### PR TITLE
Feat: treemap/basic

### PR DESCRIPTION
**Pull request by jung-han**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

#### basic
image

#### dataLabels

image

#### treemapLeaf

```js
const options = {
  series: {
    dataLabels: {
       useTreemapLeaf: boolean
    }
  }
}
```

image

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by jungeun-cho** [comment]
layout legend영역 수정되어야하지 않을까요?
스크린샷 2020-09-08 16 59 19

* 참고) 차트3
스크린샷 2020-09-08 17 00 11

**Comment by jungeun-cho** [comment]
`colorValue`들어가면서 legend영역 작업하겠네요 ㅎㅎ

File: src/store/treemapSeriesData.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  state: () => ({
+    treemapSeries: [],
+  }),
+  action: {
+    setTreemapSeriesData({ state }) {
+      const { series, theme } = state;
+
+      const treemapSeries = makeTreemapSeries(series, theme);
+
+      extend(state.treemapSeries, treemapSeries);
```
Comment (shiren) [inline src/store/treemapSeriesData.ts:102]: 중간는 따로 안써도 될것 같아요

**Review by shiren — APPROVED** [review]
수고하셨습니다.

**Review by jungeun-cho — APPROVED** [review]
[9/8] 이견없습니다. 수고하셨어요~
colorValue 들어간모습도 궁금해지네요~! :)

File: src/component/treemapSeries.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      y: 0,
+    });
+
+    const series: TreemapRectModel[] = this.combineBoundMap(seriesData, boundMap).map((m) => ({
+      ...m,
+      type: 'rect',
+    }));
+    const layer = series.map((m) => ({ ...m, color: getRGBA('#000000', m.opacity!) }));
+
+    return { series, layer };
```
[the remark this answers is not in the document]
Comment (jung-han) [reply src/component/treemapSeries.ts:121]: 차트 3의 방법을 그대로 가져와서 적용했습니다

File: src/brushes/dataLabel.ts [hunk 3]
```diff
@@ -20,6 +20,9 @@ function getStyleDefaultName(type: DataLabelType) {
   } else if (type === 'pieSeriesName') {
     styleDefault = 'pieSeriesName';
     strokeStyleDefault = 'none';
+  } else if (type === 'treemapSeriesName') {
+    styleDefault = 'treemapSeriesName';
+    strokeStyleDefault = 'none';
```
[the remark this answers is not in the document]
Comment (jung-han) [reply src/brushes/dataLabel.ts:25]: map으로 작성해서 수정했습니다.

File: src/brushes/label.ts [hunk 4]
```diff
@@ -13,7 +13,8 @@ export type LabelStyleName =
   | 'axisTitle'
   | 'stackTotal'
   | 'sector'
-  | 'pieSeriesName';
+  | 'pieSeriesName'
+  | 'treemapSeriesName';
 export type StrokeLabelStyleName = 'none' | 'stroke';
```
[the remark this answers is not in the document]
Comment (jung-han) [reply src/brushes/label.ts:18]: 기본 스타일이 달라서 다르게 가져가야 할 것 같습니다

**Pull request merged**